### DISCUSSION TITLE
Java Maven inside code-server docker container

### DISCUSSION THREAD

**Discussion by KiqoCode** [coder/code-server#3320] [Q&A]
Hi,

As the title describes it, how do I setup a code-server docker container to be able to run Java Maven projects? I'm running codecom/code-server image at the moment, but n'either java or mvn is available. 

Best Regards
Kiqo

**Comment by MxD-js** [comment]
I've battled the docker images of coder server, it's difficult to maintain the different versions of Java and Maven manually inside an image container, some images use Apine and others Ubuntu as the base image, if you can manually build an image that contains the software then it might work but I've given up and since installed Ubuntu 20.04 on my home server and installed code-server then maven and Java, now that I don't have to worry about updates overriding my separate installs, etc and the apt will update my jdk and maven as updates are issued, I don't have to do it manually. Hope this helps. But if you really want to do it then exec -it into the container and install as you normally would on any Linux distro.
